spec/models/claim/pii_scrubber_spec.rb

```diff
@@ -6,26 +6,26 @@
   it "does not delete details from a submitted claim" do
     claim = create(:claim, :submitted, updated_at: over_two_months_ago)
 
-    expect { Claim::PiiScrubber.new.scrub_completed_claims }.not_to change { claim.reload.attributes }
+    expect { Claim::PiiScrubber.new.scrub_completed_claims }.not_to change { claim.reload.attributes_before_type_cast }
```

Comment:
Looking at this it’s not clear to me why we‘ve had to switch from `attributes` to `attributes_before_type_cast`. I’m assuming that it’s because whatever notion of equality that the `change` matcher uses is struggling with the custom type. But if that’s the case, I think it would be good to validate that assumption and explain it in the commit message at least.

But, are we sure that we can’t get this working just with `attributes`? Because future developers will, I think, expect to be able to use `attributes`, and it would be unfortunate if they have to figure out each time they try to write a test like this that they need to use `attributes_before_type_cast` to make it work.

Reply:
So, yes, the specs fail when we do a straight comparison of the attributes, which I assumed was down to the use of our custom type as that was the only thing that changed, but to test the assumption more fully I've dug into the code for the matcher and it looks like it _should_ pass with the typecasting as the `RSpec::Support::FuzzyMatcher` that gets used under the hood doesn't complain about attributes with the type-cast attribute. Will dig further...

Reply:
Re the attribute type-casting, figured it out. It looks like RSpec uses `Object#hash` to do object comparison with nested objects – https://github.com/rspec/rspec-expectations/pull/1034. Defining `AcademicYear#hash` and have it return a consistent value the same academic year removes the need for this change.